e2e/test/helpers/templates/FaultInjection.cs

```diff
@@ -235,42 +220,13 @@ public static bool FaultShouldDisconnect(string faultType)
                 }
 
                 await deviceClient.CloseAsync().ConfigureAwait(false);
-
-                if (transport != Client.TransportType.Http1)
-                {
-                    if (FaultShouldDisconnect(faultType))
-                    {
-                        // 4 is the minimum notification count: connect, fault, reconnect, disable.
-                        // There are cases where the retry must be timed out (i.e. very likely for MQTT where otherwise
-                        // we would attempt to send the fault injection forever.)
-                        connectionStatusChangeCount.Should().BeGreaterOrEqualTo(4, $"For {testDevice.Id}");
-                    }
-                    else
-                    {
-                        // 2 is the minimum notification count: connect, disable.
-                        // We will monitor the test environment real network stability and switch to >=2 if necessary to
-                        // account for real network issues.
-                        connectionStatusChangeCount.Should().Be(2, $"For {testDevice.Id}");
-                    }
-                    lastConnectionStatus.Should().Be(ConnectionStatus.Disabled, $"For {testDevice.Id} with reason {lastConnectionStatusChangeReason}");
-                    lastConnectionStatusChangeReason.Should().Be(ConnectionStatusChangeReason.Client_Close);
-                }
             }
             finally
             {
                 await cleanupOperation().ConfigureAwait(false);
 
                 deviceClient.Dispose();
                 await testDevice.RemoveDeviceAsync().ConfigureAwait(false);
-
-                faultInjectionDuration.Stop();
```

Comment:
Waiting for the remaining duration doesn't seem to matter. These devices are deleted just after, so it shouldn't matter.